user-defined-plugin.md

```diff
@@ -0,0 +1,261 @@
+
+#### rSourceFile (required)
+
+`rSourceFile` holds a name of R script file that the plugin depends. Let's use `plugin.r` for this example.
+
+#### hasQueryField (optional)
+
+If you want to have a dedicated query input field that has much space for your query string, Set `hasQueryField` as `true` and add a parameter whose `name` is `query` and `dataType` is text.
+
+For example, if you define following in your `plugin.json`
+
+``` 
+"hasQueryField" : true,
+"inputParameters": [
+  {
+    "name" : "query",
+    "displayName":"Query",
+    "dataType":"text",
+    "defaultValue" : "",
+    "required" : true
+  },
+```
+then you can have dedicated query input field on right hand side of the import Dialog.
+
+![](import/images/plugin_with_query_field.png)
+
+### inputParameters
+
+`inputParameters` is an array of parameter passed to the `function` (i.e in this case, `riem_measures`) and these are rendered as input fields on Data Import Dialog. Parameters order matters so make sure to set input parameters in a way that underly R function expects. For example, if your R function has arguments station, start_date, and end_date then you need to define your inputParameters in this order.(i.e station, start_date, and end_date). If you want to define parameter order in a different way, you need to write wrapper function in library.r file and then do a parameter mapping there like below and set the wrapper function name (`riem_measures_wrapper`) to `function` attribute.
+
+```
+riem_measures_wrapper <- function(start_date, end_date, station){
+  riem_measures(station = station, start_date = start_date, end_date = end_date)  
+}
+```
+
+ Each parameter can have following Attributes
+
+- name
+- displayName
+- colSpan
+- dataType
+- showLabel
+- defaultValue
+- required
+
+#### name (required)
+
+`name` of the input parameter. 
+
+#### displayName (required)
+
+`displayName` attribute is used for Display Name on Data Source Picker Dialog and Import Data Dialog.
+
+#### colSpan (required)
+
+`colSpan` attribute holds width of the input parameter on UI. Please use 5 for now.
+
+#### dataType (required)
+
+`dataType` attribute holds a type of the input parameter. Supported types are:
+
+- text
+- select
+- number
+- boolean
+
+We do not support Date type parameter for now.
+
+##### text
+
+if you use `text`, it becomes input field that accepts characters.
+
+##### select
+
+This is useful when you want to create a static List of Values.
+For example, if you want to create a time range List of Values, you can create it by specifying `options` and `itemDataType` like below. `options` an array of selector options and each option needs to have `label` and `value` attributes.If your option value is text, `itemDataType` should be set as `text`. If your option value is number, `itemDataType` should be number. Date is not supported for `itemDataType` for now.
```

Comment:
I guess "is" is missing in the following sentence?
`options` an array of selector options and each option needs to have `label` and `value` attributes.